src/components/transition/Transition.jsx

```diff
@@ -208,8 +209,22 @@ function BaseTransition({
   const onTransitionStartRef = React.useRef();
   const onTransitionEndRef = React.useRef();
 
-  onTransitionStartRef.current = onTransitionStart;
-  onTransitionEndRef.current = onTransitionEnd;
+  React.useLayoutEffect(() => {
```

Comment:
We're gonna need isomorphic version of this effect to make it work in SSR env:

```
export const isBrowser = typeof window !== "undefined";
export const useIsomorphicLayoutEffect = isBrowser ? useLayoutEffect : useEffect;
```

Reply:
True, but I believe we should use such approach with every `useLayoutEffect` in this component.

Reply:
true, anytime we need `useLayoutEffect` we should use `useIsomorphicLayoutEffect`. The only reason we fallback to `useEffect` is that it has the same interface as `useLayoutEffect` and we're sure it won't execute on server.

Reply:
Second question, what should be the result for SSR because there won't be any transition?

- Transitions should NOT apply any styles to the element.
- Transition should apply styles of the target phase, the phase after the transition.
- Transitions without delay should invoke both `transitionStart` and `transitionEnd` callbacks synchronously.
- Transitions with delay should NOT call any callbacks.

Reply:
Ok, then I understand that it won't be a case because `useEffect` won't be executed on SSR.

Reply:
https://github.com/brainly/style-guide/pull/2332/commits/3624afa16d7a4c95b783491fc45d90a26f260fb5